### ISSUE TITLE
[`flake8-executable`] `EXE001` flags ordinary `#!` comments as shebangs

### ISSUE THREAD

**Issue by ntBre** [astral-sh/ruff#27028]
`EXE001` treats an ordinary comment beginning with `#!` as a shebang even when it appears inside a function and does not specify an interpreter.

```python
# test.py (not executable)
def f():
    #! regular comment
    return 1
```

```console
$ ruff check --isolated --select EXE001 test.py
EXE001 Shebang is present but file is not executable
 --> test.py:3:5
  |
1 | # test.py (not executable)
2 | def f():
3 |     #! regular comment
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
4 |     return 1
  |

Found 1 error.
```

This probably applies to other shebang rules too since they all share `ShebangDirective::try_extract`. I guess I assumed that that would filter out "shebang" lines later in the file.

**Comment by waterWang** [comment]
Thanks for the detailed report! I've opened a fix in #27739.

**Root cause:** `from_tokens` iterated over *all* comment ranges in the file, treating every `#!`-prefixed comment as a shebang directive. `EXE001` then fired for ordinary `#!` comments anywhere in the file.

**Fix:** Only the first comment in the file can be a shebang, and `EXE001` now only fires if that first comment starts at the beginning of the file (position 0). Other rules (EXE003/004/005) keep their existing behavior for misplaced `#!` comments.
